bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -180,6 +192,48 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 device.setMute(MuteInstructionChannel.VIDEO, muteOn);
               }
             }
+          } else {
+            logger.debug("Received unknown audio/video mute command {}", command);
+          }
+          break;
+        case CHANNEL_LAMP_ACTIVE:
+        case CHANNEL_LAMP_HOURS:
+          if (command == RefreshType.REFRESH) {
+            List<LampState> lampStates = device.getLampStatesCached();
+            // update all lamp related channels, as the response contains information about all of them
+            for (Channel lampChannel : thing.getChannels()) {
+              String channelTypeId = getChannelTypeId(lampChannel);
+              boolean isLampActiveChannel = CHANNEL_LAMP_ACTIVE.equals(channelTypeId);
+              boolean isLampHoursChannel = CHANNEL_LAMP_HOURS.equals(channelTypeId);
+
+              if (isLampActiveChannel || isLampHoursChannel) {
+                int lampNumber = ((BigDecimal) lampChannel.getConfiguration().get(CHANNEL_PARAMETER_LAMP_NUMBER))
+                    .intValue();
+                try {
+                  LampState lampState = lampStates.get(lampNumber - 1);
+                  if (isLampActiveChannel) {
+                    updateState(lampChannel.getUID(), lampState.isActive() ? OnOffType.ON : OnOffType.OFF);
+                  }
+                  if (isLampHoursChannel) {
+                    updateState(lampChannel.getUID(), new DecimalType(lampState.getLampHours()));
+                  }
+                } catch (IndexOutOfBoundsException e) {
+                  logger.debug("Status information for lamp {} is not available", lampNumber);
+                  throw new ConfigurationException("Status information for lamp " + lampNumber + " is not available");
+                }
+              }
+            }
+          } else {
+            logger.debug("Received unknown lamp state command {}", command);
+          }
+          break;
+        default:
+          // configured channels
+          Channel channel = thing.getChannel(channelUID.getId());
+          if (channel != null) {
+            this.handleCommand(new ChannelUID(getThing().getUID(), getChannelTypeId(channel)), command);
```

Comment:
This redirection looks a bit silly, I'll have a look myself if there is a clean solution to it.

Reply:
Found a solution, identifying the channel type id and having the switch based on that seems to be appropriate.